testsuite/tests/tool-toplevel/uncaught_exceptions.ml

```diff
@@ -0,0 +1,39 @@
+(* TEST
+   * expect
+*)
+
+(* PR#8594 *)
+Printexc.register_printer (fun e -> 
+  match e with 
+    | Division_by_zero -> Some "A division by zero is undefined"
+    | _ -> None);;
+Printexc.register_printer (fun e -> 
+  match e with 
+    | Exit -> Some "Catching an exit"
+    | _ -> None);;
+raise Not_found;;
+[%%expect{|
+- : unit = ()
+- : unit = ()
```

Comment:
It would be more readable if each sentence had its own expect test, instead of this "interesting" sentence having an expect-block that starts with two unrelated outputs (I thought this was an error at first).